eventcrypto.go

```diff
@@ -184,3 +184,55 @@ func verifyEventSignature(signingName string, keyID KeyID, publicKey ed25519.Pub
 
 	return VerifyJSON(signingName, keyID, publicKey, redactedJSON)
 }
+
+// VerifyEventSignatures checks that each event in a list of events has valid
+// signatures from the server that sent it.
+func VerifyEventSignatures(events []Event, keyRing KeyRing) error {
+	var toVerify []VerifyJSONRequest
+	for _, event := range events {
+		redactedJSON, err := redactEvent(event.eventJSON)
+		if err != nil {
+			return err
+		}
+		v := VerifyJSONRequest{
+			Message:    redactedJSON,
+			AtTS:       event.OriginServerTS(),
+			ServerName: event.Origin(),
+		}
+		toVerify = append(toVerify, v)
+
+		// "m.room.member" invite events are signed by both the server sending
+		// the invite and the server the invite is for.
+		if event.Type() == "m.room.member" && event.StateKey() != nil {
```

Comment:
I'm getting increasingly anxious about using hard-coded strings over constants as we continue to grow this library. It would be nice if you could declare some consts so we cannot possibly typo `m.room.memmber` - I don't expect it to be done in this PR.